dipy/segment/energy_mrf.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import division, print_function, absolute_import
+import numpy as np
+
+
+def total_energy(masked_image, masked_segmentation,
+                 mu, var, index, label, beta):
+
+    energytotal = log_likelihood(masked_image, mu, var, index, label)
+    energytotal += gibbs_energy(masked_segmentation, index, label, beta)
+
+    return energytotal
+
+
+def log_likelihood(img, mu, var, index, label):
+
+    loglike = ((img[index] - mu[label]) ** 2) / (2 * var[label])
+
+    loglike += np.log(np.sqrt(var[label]))
+
+    return loglike
+
+
+def gibbs_energy(seg, index, label, beta):
```

Comment:
Iterating over the neighbors of a voxel can be coded more compactly by defining arrays of offsets:
https://github.com/nipy/dipy/blob/master/dipy/align/sumsqdiff.pyx#L319
Then you just iterate:
https://github.com/nipy/dipy/blob/master/dipy/align/sumsqdiff.pyx#L357
adding the offsets to the coordinates of the voxel:
https://github.com/nipy/dipy/blob/master/dipy/align/sumsqdiff.pyx#L358
https://github.com/nipy/dipy/blob/master/dipy/align/sumsqdiff.pyx#L361
https://github.com/nipy/dipy/blob/master/dipy/align/sumsqdiff.pyx#L364